hack/bin/deps.bzl

```diff
@@ -125,11 +125,18 @@ def install_misc():
 
 # Install dependencies used by the controller-runtime integration test framework
 def install_integration_test_dependencies():
+    http_file(
+        name = "kube-apiserver_darwin_arm64",
+        executable = 1,
+        sha256 = "b575b687097b4662b7eb2ea5453e4b381994ddf34b85ca4bb8e4f16c5f86bedb",
+        urls = ["https://storage.googleapis.com/cockroach-operator-testing-assets/kube-apiserver-1.24.2_darwin_arm64"],
+    )
+
     http_file(
         name = "kube-apiserver_darwin_amd64",
         executable = 1,
-        sha256 = "a874d479f183f9e4c19a5c69b44955fabd2e250b467d2d9f0641ae91a82ddbea",
-        urls = ["https://storage.googleapis.com/cert-manager-testing-assets/kube-apiserver-1.17.3_darwin_amd64"],
+        sha256 = "800eedc293bcc72eb962df3e8006d7106790ca583a4ea5c107453bab3bf859b4",
+        urls = ["https://storage.googleapis.com/cockroach-operator-testing-assets/kube-apiserver-1.24.2_darwin_amd64"],
```

Comment:
We should update [kube-apiserver_linux_amd64](https://github.com/cockroachdb/cockroach-operator/pull/1041/files#diff-3e7ac4b6f3f0eb5e28e1f5bb072b149077c88cb3a67d65ed628b6ffd6986dcf9R146) as well.